### PR TITLE
[stable28] fix(sharing): Pin @nextcloud/dialogs to a version that doesn't use Nextcloud 29 APIs

### LINKED ISSUE #45888: [Bug]: Move folders/files to public link share: No user ID found
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

Moving files or folders inside a public link share (as anonymous user) failed.
The moving-window "Choose target filder" pops up but there is no content (only loading bars)

### Steps to reproduce

1. create link share with the following rights: "can edit"
2. open link share in a private window without logging in
3. moving process cannot be carried out in the private window

### Expected behavior

The contents of other folders/files will be inside the link share instead of the loading bars and it is possible to copy/move data

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

28

### Operating system

RHEL/CentOS

### PHP engine version

PHP 8.2

### Web server

Nginx

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [X] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

_No response_

### List of activated Apps

```shell
Enabled:
  - activity: 2.20.0
  - circles: 28.0.0
  - cloud_federation_api: 1.11.0
  - comments: 1.18.0
  - contactsinteraction: 1.9.0
  - dashboard: 7.8.0
  - dav: 1.29.1
  - federatedfilesharing: 1.18.0
  - federation: 1.18.0
  - files: 2.0.0
  - files_pdfviewer: 2.9.0
  - files_reminders: 1.1.0
  - files_sharing: 1.20.0
  - files_trashbin: 1.18.0
  - files_versions: 1.21.0
  - firstrunwizard: 2.17.0
  - logreader: 2.13.0
  - lookup_server_connector: 1.16.0
  - nextcloud_announcements: 1.17.0
  - notifications: 2.16.0
  - oauth2: 1.16.3
  - password_policy: 1.18.0
  - photos: 2.4.0
  - privacy: 1.12.0
  - provisioning_api: 1.18.0
  - recommendations: 2.0.0
  - related_resources: 1.3.0
  - serverinfo: 1.18.0
  - settings: 1.10.1
  - sharebymail: 1.18.0
  - support: 1.11.1
  - survey_client: 1.16.0
  - systemtags: 1.18.0
  - text: 3.9.1
  - theming: 2.3.0
  - twofactor_backupcodes: 1.17.0
  - updatenotification: 1.18.0
  - user_ldap: 1.19.0
  - user_saml: 6.1.3
  - user_status: 1.8.1
  - viewer: 2.2.0
  - weather_status: 1.8.0
  - workflowengine: 2.10.0
```

### Nextcloud Signing status

_No response_

### Nextcloud Logs

```shell
All logs to sample Link-Share: N6zoA28bzBmq7ta

Jun 15, 202413:37:25.877

Jun 15 13:37:25 nctest01 nctest: myclientIP - N6zoA28bzBmq7ta [15/Jun/2024:13:37:25 +0200] "PROPFIND /public.php/webdav// HTTP/2.0" 207 3907 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/115.0" "TLSv1.3/TLS_AES_256_GCM_SHA384" "-"

source: nctest01 event_type: v4_f6f01d98 facility: local7 priority: info hostname: nctest01 appname: nctest
Jun 15, 202413:37:25.869

Jun 15 13:37:25 nctest01 nctest: myclientIP - N6zoA28bzBmq7ta [15/Jun/2024:13:37:25 +0200] "PROPFIND /public.php/webdav// HTTP/2.0" 207 3907 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/115.0" "TLSv1.3/TLS_AES_256_GCM_SHA384" "-"

source: nctest01 event_type: v4_f6f01d98 facility: local7 priority: info hostname: nctest01 appname: nctest
Jun 15, 202413:37:21.528

Jun 15 13:37:21 nctest01 nctest: myclientIP - N6zoA28bzBmq7ta [15/Jun/2024:13:37:21 +0200] "PROPFIND /public.php/webdav/ HTTP/2.0" 207 3347 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/115.0" "TLSv1.3/TLS_AES_256_GCM_SHA384" "-"

source: nctest01 event_type: v4_f6f01d98 facility: local7 priority: info hostname: nctest01 appname: nctest
Jun 15, 202413:37:21.511

Jun 15 13:37:21 nctest01 nctest: myclientIP - - [15/Jun/2024:13:37:21 +0200] "GET /ocs/v2.php/apps/text/public/workspace?path=%2F&shareToken=N6zoA28bzBmq7ta HTTP/2.0" 404 112 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/115.0" "TLSv1.3/TLS_AES_256_GCM_SHA384" "-"

source: nctest01 event_type: v4_2d35fd7c facility: local7 priority: info hostname: nctest01 appname: nctest
Jun 15, 202413:37:11.658

Jun 15 13:37:11 nctest01 nctest: myclientIP - - [15/Jun/2024:13:37:11 +0200] "GET /s/N6zoA28bzBmq7ta HTTP/2.0" 200 8324 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/115.0" "TLSv1.3/TLS_AES_256_GCM_SHA384" "-"
```

### Additional info

Error in Browser-Console:

```
Error: No user id found
    M index.mjs:871
    l FilePicker-DBGB1Rec.mjs:797
    f FilePicker-DBGB1Rec.mjs:825
core-common.js:2:2116787
    cn https://nctest/dist/core-common.js?v=8453725f-0:2
    dn https://nctest/dist/core-common.js?v=8453725f-0:2
    ln https://nctest/dist/core-common.js?v=8453725f-0:2
    un https://nctest/dist/core-common.js?v=8453725f-0:2
    ```

### PR DESCRIPTION
**Pull request by nickvergessen**
* Resolves: #45888

## Summary

1. Create a folder with subfolder
2. Create a public link
3. Open public link in private Tab
4. Upload a file
5. Use the checkbox and the `… Action`
6. Try to move to the subfolder

Before | After
---|---
grafik | grafik
grafik | grafik

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by ShGKme — APPROVED** [review]
I cannot find the corresponding PR, but I remember that bump fixed 3 other issues.

**Comment by nickvergessen** [comment]
- 28 is in https://github.com/nextcloud/server/pull/46422
- Backported from 29 which is in https://github.com/nextcloud/server/pull/46374
- fixes in there are linked to a support issue from the same customer, then I think we have to go https://github.com/nextcloud-libraries/nextcloud-dialogs/issues/1418

**Comment by susnux** [comment]
I created the PRs for branching v6.0.0 on the dialogs library and reverted that problematic patch, please review.
After that we can update stable28 to v5.3.6 and stable29 / stable30 / master to v6.0.0

**Review by skjnldsv — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
     "@nextcloud/browserslist-config": "^3.0.1",
     "@nextcloud/calendar-availability-vue": "^2.2.2",
     "@nextcloud/capabilities": "^1.2.0",
-    "@nextcloud/dialogs": "^5.3.5",
+    "@nextcloud/dialogs": "5.3.2",
```
Comment (susnux) [inline package.json:48]: Can be unpinned now, but use 5.3.6 or later

```suggestion
    "@nextcloud/dialogs": "^5.3.6",
```

**Review by susnux — APPROVED** [review]

**Comment by skjnldsv** [comment]
`files_copy-move.cy.ts` doesnt't look so good :confused: 
Will restart

**Comment by nickvergessen** [comment]
It had failed before your rebase and I had restarted it.
So might be related

**Comment by susnux** [comment]
It is related. This is a bug that was fixed but now back with the revert.

**Comment by susnux** [comment]
Fix: https://github.com/nextcloud-libraries/nextcloud-dialogs/pull/1428

**Pull request merged**